sdcm/wait.py

```diff
@@ -53,18 +53,27 @@ def retry_logger(retry_state):
             retry_state.attempt_number,
             retry_state.outcome._exception or retry_state.outcome._result,
         )
+    stops = [tenacity.stop_after_delay(timeout)]
+    if stop_event := kwargs.pop("stop_event", None):
+        stops.append(tenacity.stop.stop_when_event_set(stop_event))
 
     try:
         retry = tenacity.Retrying(
             reraise=throw_exc,
-            stop=tenacity.stop_after_delay(timeout),
+            stop=tenacity.stop_any(*stops),
             wait=tenacity.wait_fixed(step),
             before_sleep=retry_logger,
             retry=(tenacity.retry_if_result(lambda value: not value) | tenacity.retry_if_exception_type())
         )
         res = retry(func, **kwargs)
 
     except Exception as ex:  # pylint: disable=broad-except
+        if stop_event and stop_event.is_set():
+            err = f": {text or func.__name__}: stopped by Event"
+            if hasattr(ex, 'last_attempt'):
+                res = ex.last_attempt._result  # pylint: disable=protected-access,no-member
+            return res
```

Comment:
why returning and not raising error as `throw_exc` might suggest ?

Reply:
if wait_for is stop by stop stopped event, it could be done only explicitly via Event.set(), and there is no need to raise the Exception, just check the result from last retry if it will be required.

Reply:
Then, to keep the interface be sane, need to rename that param to something like `throw_exc_on_timeout`.

Reply:
Yes, interface will be changed. And yes, it will require updates to the existing explicit usages of it by it's consumers.
It is smaller problem than it is now - `not throwing exception when it is explicitely set to be "True"`.

Reply:
@fruch , @temichus WDYT?

Reply:
@vponomaryov ,  i preview how many changes is required for renaming, i decided to raise exception on if throw_exc true and event set. Is it ok?

Reply:
> @vponomaryov , i preview how many changes is required for renaming, i decided to raise exception on if throw_exc true and event set. Is it ok?

Ok. But I don't see it in the PR's code changes.

Reply:
You can see, that in Exception block if event was set , i just change the error message, and if throw_exc is true, then it will be raised with another error message